third_party/kmod/CMakeLists.txt

```diff
@@ -81,3 +80,66 @@ add_custom_command(
         ""
     VERBATIM
 )
+else()
+message(STATUS "Skipping Meltdown kernel module on non-x86/64/ppc CPUs")
+endif()
+
+if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "^aarch64$")
+set(MODULE_NAME eret_hvc_smc_module)
+
+# Move the inputs into a directory in the output first. Kbuild seems to use
+# KBUILD_EXTMOD to set both the source of the out-of-tree kernel module to
+# build *and* the build output directory, so we can't build from the source
+# tree while putting outputs somewhere else.
+set(MODULE_OUTPUT_DIR ${CMAKE_BINARY_DIR}/${MODULE_NAME})
+file(MAKE_DIRECTORY ${MODULE_OUTPUT_DIR})
+file(COPY Makefile.eret_hvc_smc ${MODULE_NAME}.c DESTINATION ${MODULE_OUTPUT_DIR})
```

Comment:
if we have different source files and different `Makefile`s it is probably nicer to make folders for two different kernel modules.